### PR TITLE
Try to add double-jump back.

### PR DESCRIPTION
**Pull request by Ekdohibs**
This still needs a lot of testing.

### CODE REVIEW

File: src/localplayer.cpp [hunk 1]
```diff
@@ -450,7 +450,7 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 	*/
 	const ContentFeatures &f = nodemgr->get(map->getNodeNoEx(getStandingNodePos()));
 	// Determine if jumping is possible
-	m_can_jump = touching_ground && !in_liquid;
+	m_can_jump = (touching_ground || (physics_override_sneak_glitch && m_sneak_node_exists && control.sneak)) && !in_liquid;
```
Comment (sfan5) [inline src/localplayer.cpp]: i'm fairly certain this would make the line too long

**Comment by Fixer-007** [comment]
Tried it few times, what I noticed...
+ shift+space increases jump height - can jump over two blocks (even if you simply jump)
+ if you place trapdoor or snow nodebox - can jump over three blocks
default

Had pretty good results in climbing 180 nodes forestry cliff. Tried some evasions - from what I seen it should not add more evasion, at least from first look, I only see damage evasion as described in https://github.com/minetest/minetest/issues/329 and evasion during special kind of landing on edges from medium height. My main conclusion, even with this I would not risk jumping down, high chance of damage and unpredictable chance of damage evasion.

I will play with this on actual servers.

**Comment by paramat** [comment]
> shift+space increases jump height - can jump over two blocks (even if you simply jump vertically,

Do you mean a 2-node jump on flat ground away from any obstacles?
A 2 node jump from just sneak+jump (next to an obstacle) is not a problem, a direction key should not be necessary.

**Comment by Fixer-007** [comment]
>Do you mean a 2-node jump on flat ground away from any obstacles?

yes

**Comment by paramat** [comment]
But in the video you are near a ledge or near a wall? I agree 2-node jump should only work if near a ledge 2 nodes up.

**Comment by Fixer-007** [comment]
@paramat does not matter, it jumps the same on completely flat ground, shift enables bunny mode

**Comment by paramat** [comment]
Ok that's not good then, we may need another method.

**Comment by paramat** [comment]
Tested, confirmed jump becomes 0.8 nodes higher even when away from any nearby nodes :-1:

**Comment by paramat** [comment]
I have a commit that might provide 2-node jump near a ledge paramat@236cde5
But has had very little testing. Does not make jump higher away from nodes (as Ekdohib's PR does).

**Comment by paramat** [comment]
Interestingly, my commit is all that is needed to make corner sneak ladder work, however simple sneak ladder now doesn't work at all (but was buggy anyway). So this is all that is needed, a corner sneak ladder is treated as a series of ledge pull-ups:
```
// pos is the node the player is standing inside(!)
static bool detectSneakLadder(Map *map, INodeDefManager *nodemgr, v3s16 pos)
{
	// Detects a structure known as "sneak ladder" or "sneak elevator"
	// that relies on bugs to provide a fast means of vertical transportation,
	// the bugs have since been fixed but this function remains to keep it working.
	// NOTE: This is just entirely a huge hack and causes way too many problems.
	bool is_valid_position;
	MapNode node;
	// X/Z vectors for 4 neighboring nodes
	static const v2s16 vecs[] = { v2s16(-1, 0), v2s16(1, 0), v2s16(0, -1), v2s16(0, 1) };

	for (u16 i = 0; i < ARRLEN(vecs); i++) {
		const v2s16 vec = vecs[i];

		// detect ledge for 2-node jump
		node = GETNODE(map, pos, vec, 1, &is_valid_position);
		if (is_valid_position && nodemgr->get(node).walkable) {
			// ledge exists
			node = GETNODE(map, pos, vec, 2, &is_valid_position);
			if (is_valid_position && !nodemgr->get(node).walkable)
				// space above ledge exists
				return true;
		}
	}

	return false;
}
```

**Comment by paramat** [comment]
New PR #5476

**Comment by paramat** [comment]
Closing for reasons above.

**Pull request closed without merging**